CYF-ITP-Sheffield|Kazeem Hamzat|Module-User-Focused-Data|Week1

HamzatKazeem opened the pull request:
<!--

You must title your PR like this:

COHORT_NAME | FIRST_NAME LAST_NAME | REPO_NAME | WEEK

For example,

NW4 | Carol Owen | HTML-CSS-Module | Week1

Complete the task list below this message.
If your PR is rejected, check the task list.

-->

## Learners, PR Template

Self checklist

- [ ] I have committed my files one by one, on purpose, and for a reason
- [ ] I have titled my PR with COHORT_NAME | FIRST_NAME LAST_NAME | REPO_NAME | WEEK 
- [ ] I have tested my changes
- [ ] My changes follow the [style guide](https://syllabus.codeyourfuture.io/guides/code-style-guide/)
- [ ] My changes meet the [requirements](./README.md) of this task

## Changelist

Briefly explain your PR.

## Questions

Ask any questions you have for your reviewer.

HamzatKazeem committed:
Initial Commit

Setting up the page skeleton and the header

HamzatKazeem committed:
Completing the first article

Completing the section of the first article

HamzatKazeem committed:
Setting up Articles 2 and 3

Setting up the containers for  Articles 2 and 3

HamzatKazeem committed:
Completing the Articles sections

Completing the sections for Articles 1, 2 and 3

HamzatKazeem committed:
Add Footer Section

Create the footer section

HamzatKazeem committed:
Add Hover Effect to Buttons

Add Hover Effect to Buttons

HamzatKazeem committed:
Change Text Color for Hover Effects

Change Text Color for Hover Effects

HamzatKazeem committed:
Removed Redundant File

Removed Redundant File

islam95 commented:
Hi @HamzatKazeem Well done for making your first PR.

Nice work. You can see your work deployed on a PR preview here:

https://deploy-preview-167--cyf-ufd.netlify.app/wireframe/

A couple of things to think about for you
- This doesn't seem to match the wireframe layout very closely. What do you think?
- Also, try to check https://validator.w3.org for suggestions to improve your html code.

islam95 commented on Wireframe/index.html:
```diff
@@ -1,13 +1,70 @@
+      <p>
+        Git is a version control software that allows developers to create and manage different versions of a project. It is a free and open source distributed version control system designed to handle everything from small to very large projects with speed and efficiency.
+      </p>
+      <a href="https://www.atlassian.com/git/tutorials/what-is-git" target="_blank"><button class="btn-format">Read More</button></a>
```
Nice. I see you used target="_blank" to open page in a new tab!

One issue here is that it's not good idea to put <button> tag inside <a> or vice versa. If you check https://validator.w3.org this will be suggested for you.
Instead try to use only <a>, and style it like a button in your css.

islam95 commented on Wireframe/css/style.css:
```diff
@@ -0,0 +1,76 @@
+  padding-top: 5px;
+  line-height: 25px;
+}
+footer{
```
How would you fix footer to the bottom of the viewport?

HamzatKazeem committed:
Implement PR Feedback

-Resolve error flagged by checked validator
-Reduce the content of header.
-Replace button within anchor tag to CCS defined button

HamzatKazeem committed:
Fixed Footer to Bottom Page

Fixed Footer to Bottom Page

HamzatKazeem commented:
@islam95 Thank you for your time and valuable feedback on my PR. I have now fixed all the recommendations. Can I now append a complete label to it? Thanks.

islam95 commented on Wireframe/css/style.css:
```diff
@@ -0,0 +1,88 @@
+  text-align: justify;
+}
+
+#footer {
```
It looks like you changed this to id="footer" which doesn't exist in your html.

islam95 commented on Wireframe/css/style.css:
```diff
@@ -0,0 +1,88 @@
+}
+
+
+a.button {
```
well done!

islam95 reviewed (changes requested):
Great work. But let's fix some other issues. 
- Footer is not fixed yet.
- Let's make the layout look more like a Wireframe picture. The combined width of your last 2 articles should be the same as the first article (which is 100% of a container). Would be able to fix that? I can give you a small hint: You may start by removing the `width: 991px` of `#other-article-title`

islam95 commented on Wireframe/css/style.css:
```diff
@@ -0,0 +1,88 @@
+  display: flex;
+  width: 991px;
+  margin:5px;
+  border: 0.1px solid white;
```
it's better to use whole numbers here instead of decimals. `0.1px` wide border would not be visible well.

HamzatKazeem committed:
Fixing PR Feedback 2

-Fixed footer to fixed position
-Adjust Articles 2 and 3 Display position
-Change values for border thickness from decimal to whole number.

HamzatKazeem committed:
Reduce Footer font size

Reduce Footer font size

HamzatKazeem committed:
Further reducing footer text size

Further reducing footer text size

HamzatKazeem commented:
Hello, thank you for your feedback, I have further fixed the issue as you advised. Thanks.

islam95 commented on Wireframe/css/style.css:
```diff
@@ -0,0 +1,97 @@
+  display: flex;
+}
+
+#other-article-title{
```
This is a container which has 2 articles in it, and not title. What would you change the name to?

HamzatKazeem replied:
Resolved by changing to: other-articles-container

islam95 commented on Wireframe/css/style.css:
```diff
@@ -0,0 +1,97 @@
+  height: 500px;
+  margin:5px;
+  border: 1px solid white;
+  border-radius: 15px;
```
Border and border-radius makes it look a bit odd. I would suggest to remove it but wanna know what is the reason behind adding it?

HamzatKazeem replied:
**Border-radius:**
I have removed the border radius since the edge of the wireframe is square-shaped. However, I use it to create a round effect at the edge of the border.
**Border:**
I retained the border to create a style effect on the page. The text will appear to be on a frame or canvas. The colour is now very similar to the background colour hoping that removes the skeletal line that was making it look odd. 
Overall, using the border helps when troubleshooting with dev tools(Inspect in Chrome)

islam95 commented on Wireframe/css/style.css:
```diff
@@ -0,0 +1,97 @@
+ 
+}
+article{
+  padding-left: 20px;
```
It will look better if you add padding not only to the left, but to all 4 sides. Let's try it, shall we?

islam95 reviewed (changes requested):
Good job! I have left few minor suggestions for you.

And one more thing is that, you have used a `first-child` `pseudo class` on article like so `article:first-child` and it applied the styles to the first and second articles on your page. Why do you think is the case and how would you fix it, so that the style will apply only on first article as intended ?

HamzatKazeem committed:
Fixing PR Feedback Issues

-Remove Border Radius
-Add padding to around articles content
-Rename CSS class to reflect it purpose

islam95 reviewed (commented):
Hi @HamzatKazeem. 
Great job! I'm really pleased with your progress and how you addressed my suggestions—you've done excellent work.

You can go ahead and complete the PR now. However, there's one additional thing you might consider, though it's entirely optional. To make the page more closely match the wireframe, you could add images to all three articles, as shown in the design. This would be a great opportunity to push yourself further, if you're looking to take your work to the next level.

HamzatKazeem committed:
Add images to Articles two and three

Add images to Articles two and three

shieldo reviewed (approved):
I can see you've added these relevant images into the page here - good work!

HamzatKazeem commented:
@islam95 Thank you for your time. I have added the images as you suggested.

HamzatKazeem committed:
Remove files for feature/controlforms branch

Remove files for feature/controlforms branch

shieldo reviewed (approved):
As above (now I can approve PRs :) )

islam95 reviewed (approved):
Well done!
